pkg/apiclient/apiclient.go

```diff
@@ -392,15 +392,13 @@ func (c *client) refreshAuthToken(localCfg *localconfig.LocalConfig, ctxName, co
 	if err != nil {
 		return err
 	}
-	parser := &jwt.Parser{
-		ValidationHelper: jwt.NewValidationHelper(jwt.WithoutClaimsValidation(), jwt.WithoutAudienceValidation()),
-	}
-	var claims jwt.StandardClaims
+	parser := jwt.NewParser(jwt.WithoutClaimsValidation())
+	var claims jwt.RegisteredClaims
```

Comment:
I was wondering why we previously skipped audience validation, and now validate audience field?

Reply:
Oh, thanks for reminding me of this since I meant to double check on this change as well! This is only because `jwt.WithoutAudienceValidation()` is no longer available in the new version of JWT client. I am not aware of the reason why the audience validation was skipped previously. If you have any suggestions here, please let me know!

Reply:
It is important to not validate audience (see https://github.com/argoproj/argo-cd/discussions/5253 and https://github.com/argoproj/argo-cd/pull/5413). Apparently, https://github.com/golang-jwt/jwt does not validate the audience automatically , so it works just as we need right now.

They might re-introduce automatic validate in future https://github.com/golang-jwt/jwt/issues/16 but we are good for now.